### ISSUE TITLE
Segmentation Fault

### ISSUE THREAD

**Issue by Salmation** [bshao001/ChatLearner#27]
Hi! 
I've been trying to run this code but I keep getting an error as follows when i run bottrainer.py:

......
/job:localhost/replica:0/task:0/device:CPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/cell_0/gru_cell/concat_1/axis: (Const): /job:localhost/replica:0/task:0/device:CPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/cell_0/gru_cell/split/split_dim: (Const): /job:localhost/replica:0/task:0/device:CPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/cell_0/gru_cell/Const: (Const): /job:localhost/replica:0/task:0/device:CPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/cell_0/gru_cell/concat/axis: (Const): /job:localhost/replica:0/task:0/device:CPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/dropout/random_uniform/max: (Const): /job:localhost/replica:0/task:0/device:CPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/dropout/random_uniform/min: (Const): /job:localhost/replica:0/task:0/device:CPU:0
dynamic_seq2seq/encoder/rnn/while/rnn/multi_rnn_cell/cell_0/dropout/keep_prob: (Const): /job:localhost/replica:0/task:0/device:CPU:0
dynamic_seq2seq/decoder/LuongAttention/concat/axis: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Adam/value: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_16: (Const): /job:localhost/replica:0/task:0/device:CPU:0
num_parallel_calls_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_15: (Const): /job:localhost/replica:0/task:0/device:CPU:0
num_parallel_calls: (Const): /job:localhost/replica:0/task:0/device:CPU:0
seed2: (Const): /job:localhost/replica:0/task:0/device:CPU:0
seed: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_3: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_14: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_13: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_12: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_11: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_10: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_9: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_8: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_7: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_6: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_2: (Const): /job:localhost/replica:0/task:0/device:CPU:0
compression_type_2: (Const): /job:localhost/replica:0/task:0/device:CPU:0
filenames_2: (Const): /job:localhost/replica:0/task:0/device:CPU:0
count_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_5: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_4: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
compression_type_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
filenames_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
count: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_3: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_2: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size: (Const): /job:localhost/replica:0/task:0/device:CPU:0
compression_type: (Const): /job:localhost/replica:0/task:0/device:CPU:0
filenames: (Const): /job:localhost/replica:0/task:0/device:CPU:0
hash_table/Const: (Const): /job:localhost/replica:0/task:0/device:CPU:0
hash_table: (HashTableV2): /job:localhost/replica:0/task:0/device:CPU:0
Const_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const: (Const): /job:localhost/replica:0/task:0/device:CPU:0
string_to_index/hash_table/table_init/asset_filepath: (Const): /job:localhost/replica:0/task:0/device:CPU:0
string_to_index/hash_table/Const: (Const): /job:localhost/replica:0/task:0/device:CPU:0
string_to_index/hash_table: (HashTableV2): /job:localhost/replica:0/task:0/device:CPU:0
Segmentation fault

Any idea what might be causing this?

I'll be really grateful for any help. 

Thanks

**Comment by bshao001** [comment]
@Salmation 
What is the batch size you are using? 

I am not sure the real cause, but I suggest you start with a very small batch size, such as 16. If it works for a few epochs, you can then increase it gradually. BTW, without a good GPU, the training will be very time-consuming. It may take you more than 1 week if you use the whole Papaya data set.

**Comment by Salmation** [comment]
Thanks for the quick response.

Really sorry to bother you, but I'm kinda newbie at this. Can you tell me how I can configure/set a custom batch size?

**Comment by bshao001** [comment]
No problem. Change it in this file: /Data/Corpus/hparams.json

**Comment by Salmation** [comment]
Thank you very much! I tried with a smaller batch size but still had the same error. Have you tried running this with a CPU? or you've only used a GPU so far?

**Comment by bshao001** [comment]
Yes. The code works with CPU as well. I have just tried with CPU, and here is what I saw:

... (skipped a lot here)
compression_type_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
filenames_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
count: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_3: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_2: (Const): /job:localhost/replica:0/task:0/device:CPU:0
buffer_size: (Const): /job:localhost/replica:0/task:0/device:CPU:0
compression_type: (Const): /job:localhost/replica:0/task:0/device:CPU:0
filenames: (Const): /job:localhost/replica:0/task:0/device:CPU:0
hash_table/Const: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const_1: (Const): /job:localhost/replica:0/task:0/device:CPU:0
Const: (Const): /job:localhost/replica:0/task:0/device:CPU:0
string_to_index/hash_table/table_init/asset_filepath: (Const): /job:localhost/replica:0/task:0/device:CPU:0
string_to_index/hash_table/Const: (Const): /job:localhost/replica:0/task:0/device:CPU:0
# Training loop started @ 2018-01-23 15:58:19

**Comment by Salmation** [comment]
It does start running but after a few seconds it breaks with this message. I've been trying to locate what's causing this but there's no real error description which can point me in the right direction :(

Anyways, I'll keep looking for clues. Thank you very much for the help. Much appreciated!

**Comment by bshao001** [comment]
Maybe you can try to uncomment those comments in tokenizeddata.py, and try to run it. And try other code may help you identify the problem. At the same time, verify your TensorFlow installation and version are correct.

BTW, my training is still running without any problem, at least for 7 minutes now.

**Comment by Salmation** [comment]
I tried running tokenizeddata.py by uncommenting the last part and it worked. But I'm still getting the error when I run the bottrainer.py file. I'll try this with another system to see if I can get it to work there. 

Thanks for all the help!
